schnorr_fun/src/frost/shared_key.rs

```diff
@@ -182,6 +182,47 @@ impl<T: PointType, Z: ZeroChoice> SharedKey<T, Z> {
     pub fn share_image(&self, index: ShareIndex) -> Point<NonNormal, Public, Zero> {
         poly::point::eval(&self.point_polynomial, index)
     }
+
+    /// Checks if the polynomial coefficients contain a fingerprint by verifying that
+    /// each coefficient (when hashed with all previous coefficients) ends with the
+    /// specified fingerprint bytes.
+    ///
+    /// This is useful for detecting if shares come from the same DKG session or if
+    /// they have been corrupted.
+    ///
+    /// ## Parameters
+    /// - `fingerprint`: The expected fingerprint bytes
+    ///
+    /// ## Returns
+    /// `true` if all coefficients match the fingerprint pattern, `false` otherwise
+    pub fn check_fingerprint<H: crate::fun::hash::Hash32>(&self, fingerprint: &[u8]) -> bool {
+        use crate::fun::hash::HashAdd;
+
+        if self.point_polynomial.is_empty() || fingerprint.is_empty() {
+            return false;
+        }
+
+        // Start with hash of just the public key
+        let mut hash_state = H::default().add(&self.point_polynomial[0]);
+
+        // Check each non-constant coefficient
+        for i in 1..self.point_polynomial.len() {
+            let hash = hash_state.clone().add(&self.point_polynomial[i]);
```

Comment:
I think the algorithm can be structured without adding this point to the hash twice (see also the one on 221).